adc/microphone_adc/microphone_adc.c

```diff
@@ -0,0 +1,48 @@
+/**
+ * Copyright (c) 2021 Raspberry Pi (Trading) Ltd.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#include <stdio.h>
+#include "pico/stdlib.h"
+#include "hardware/gpio.h"
+#include "hardware/adc.h"
+#include "hardware/uart.h"
+#include "pico/binary_info.h"
+
+ /* Example code to extract analog values from a microphone using the ADC
+    with accompanying Python file to plot these values
+
+    Connections on Raspberry Pi Pico board, other boards may vary.
+
+    GPIO 26/ADC0 (pin 31)-> AOUT or AUD on microphone board
+    3.3v (pin 36) -> VCC on microphone board
+    GND (pin 38)  -> GND on microphone board
+ */
+
+#define ADC_PIN 0
+#define ADC_VREF 3.3
+#define ADC_RANGE (1 << 12)
+#define ADC_CONVERT ADC_VREF / ADC_RANGE
```

Comment:
Should this be `#define ADC_CONVERT ADC_VREF / (ADC_RANGE - 1)` ?

Reply:
I think that's what I had originally, but there are 4096 different voltage values the output could take, right?

Reply:
Yup, but one of those 4096 values is zero :wink:  I have to confess, I always find this confusing too!

So, the ADC can read values between 0V (which will be a 12-bit 0) and 3.3V (which will be a 12-bit 4095).
Which means we want a reading of 4095 from the ADC to come back out again as 3.3V.

With `#define ADC_CONVERT ADC_VREF / ADC_RANGE`, an ADC value of 4095 would result in a human-readable voltage of 3.299V.
With `#define ADC_CONVERT ADC_VREF / (ADC_RANGE - 1)`, an ADC value of 4095 would result in a human-readable voltage of 3.3V.

(but of course for all practical purposes the difference here will actually be indistinguishable :laughing: )

Reply:
Ah!